test/sqllogictest/window_funcs.slt

```diff
@@ -360,6 +356,37 @@ WITH t (x) AS (VALUES ('a'), ('b'), ('c'))
 SELECT * FROM t
 WHERE row_number() over () > 1;
 
+statement ok
+CREATE TABLE not_allowed_tests (
+  v INT,
+  w INT,
+  k INT
+);
+
+query error window functions are not allowed in GROUP BY
+SELECT * FROM not_allowed_tests GROUP BY v, count(w) OVER ();
+
+query error window functions are not allowed in GROUP BY
+SELECT count(w) OVER () FROM not_allowed_tests GROUP BY 1;
+
+query error window functions are not allowed in RETURNING
+INSERT INTO not_allowed_tests (k, v) VALUES (99, 100) RETURNING sum(v) OVER ();
+
+query error LIMIT must be an integer constant
+SELECT sum(v) FROM not_allowed_tests GROUP BY k LIMIT sum(v) OVER ();
+
+query error OFFSET must be an integer constant
+SELECT sum(v) FROM not_allowed_tests GROUP BY k LIMIT 1 OFFSET sum(v) OVER ();
+
+query error window functions are not allowed in VALUES
+INSERT INTO not_allowed_tests (k, v) VALUES (99, count(1) OVER ());
+
+query error window functions are not allowed in WHERE
+SELECT k FROM not_allowed_tests WHERE avg(k) OVER () > 1;
+
+query error window functions are not allowed in HAVING
+SELECT 1 FROM not_allowed_tests GROUP BY 1 HAVING sum(1) OVER (PARTITION BY 1) > 1;
+
```

Comment:
Lifted these from `cockroach/window.slt`, which is currently an ignored file. (I'm planning to open a follow-up PR that enables that file, as now large parts of it have become supported by us too.)